flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -89,17 +101,98 @@ public void checkpointState(
 			checkpointMetaData.getCheckpointId(),
 			checkpointOptions.getTargetLocation());
 
-		CheckpointingOperation.execute(
-			checkpointMetaData,
-			checkpointOptions,
-			checkpointMetrics,
-			storage,
-			operatorChain,
-			taskName,
-			closeableRegistry,
-			executorService,
-			env,
-			asyncExceptionHandler,
-			isCanceled);
+		long startSyncPartNano = System.nanoTime();
```

Comment:
nit: better to extract a separate method for the following operation, otherwise this method seems too long for not easy tracing the steps.